src/include/sof/math/trig.h

```diff
@@ -137,17 +159,151 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
+ * then a multiple of pi/2 is added or subtracted from the
+ * angle until it is within the range [-pi/2,pi/2).Start
+ * with the angle in the range [-2*pi, 2*pi) and output has
+ * range in [-1.0 to 1.0]
+ * +------------------+-----------------+--------+------------+
+ * | thRadFxp	      |cdccexpth        |thRadFxp|   cdccexpth|
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * | 32 | 28  |  1    | 32 | 15 |   1	| 4.28	 | 17.15       |
+ * +------------------+-----------------+--------+------------+
+ */
+static inline struct cmpx cmpx_exp_16b(int32_t th_rad_fxp)
+{
+	int32_t th_cdc_fxp;
+	int32_t sign;
+	int32_t b_yn;
+	int32_t xn;
+	struct cmpx cexp;
+
+	cordic_cfg type = EN_16B_CORDIC_CEXP;
+	/* compute coeff from angles */
+	cordic_approx(th_rad_fxp, type, &sign, &b_yn, &xn, &th_cdc_fxp);
+	cexp = cmpx_cexp(sign, b_yn, xn, type);
+	/* return the complex result */
+	return cexp;
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * Inverse cosine angle values in rad
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 31
+ * Error (max = 0.000000026077032), THD+N  = -157.948952635422842
+ */
+static inline int32_t acosine_fixed_32b(int32_t cdc_acos_th)
+{
+	int32_t saturated_unary_minus;
+	int32_t th_rad_fxp;
+	cordic_cfg type = EN_32B_CORDIC_ACOSINE;
+
+	saturated_unary_minus = cdc_acos_th;
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic(saturated_unary_minus, CORDIC_31B_TABLE_SIZE,
+					      cordic_ilookup, type);
+	} else {
+		if (saturated_unary_minus <= INT32_MIN)
+			saturated_unary_minus = INT32_MAX;
+		else
+			saturated_unary_minus = -saturated_unary_minus;
+
+	    th_rad_fxp =
+		PI_MUL2_Q4_28 - is_scalar_cordic(saturated_unary_minus, CORDIC_31B_TABLE_SIZE,
+						 cordic_ilookup, type);
+	}
+	return th_rad_fxp;
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th,input value range [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 31
+ * Error (max = 0.000000027939677), THD+N  = -157.454534077921551
+ */
+static inline int32_t asin_fixed_32b(int32_t cdc_asin_th)
+{
+	int32_t saturated_unary_minus;
+	int32_t th_rad_fxp;
+	cordic_cfg type = EN_32B_CORDIC_ASINE;
+
+	saturated_unary_minus = cdc_asin_th;
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic(saturated_unary_minus, CORDIC_31B_TABLE_SIZE,
+					      cordic_ilookup, type);
+	} else {
+		if (saturated_unary_minus <= INT32_MIN)
+			saturated_unary_minus = INT32_MAX;
+		else
+			saturated_unary_minus = -saturated_unary_minus;
+
```

Comment:
Can this statement be
```c
if (saturated_unary_minus >= 0) {
    do_this();
} else {
    saturated_unary_minus = -saturated_unary_minus;
}
```

Reply:
Yes, we can delete below check for both arc cosine and arc sine functions. 

Since input range for arc cosine is [-1 to +1]
```
if (saturatedUnaryMinus <= INT32_MIN) {
                                saturatedUnaryMinus = INT32_MAX;
                        }

```
And can do like below
```
                if (saturatedUnaryMinus >= 0) {
                        th_rad_fxp[i] = iScalarCordicACos(saturatedUnaryMinus, (int16_t)31, iv);
                } else {
                        saturatedUnaryMinus = -saturatedUnaryMinus;
                        th_rad_fxp[i] =
                            1686629713 - iScalarCordicACos(saturatedUnaryMinus, (int16_t)31, iv);
                }

```